GCSViews/FlightPlanner.cs

```diff
@@ -1,8008 +1,8124 @@
-                gp.AddString(text, font.FontFamily, (int) font.Style, font.Size, new PointF(0, 0), sf);
-                tr.Rotate(float.Parse(rotation));
-                gp.Transform(tr);
-
-                utmpos basepos = new utmpos(MouseDownStart);
-
-                try
-                {
-
-                    foreach (var pathPoint in gp.PathPoints)
-                    {
-                        utmpos newpos = new utmpos(basepos);
-
-                        newpos.x += pathPoint.X;
-                        newpos.y += -pathPoint.Y;
-
-                        var newlla = newpos.ToLLA();
-                        quickadd = true;
-                        AddWPToMap(newlla.Lat, newlla.Lng, int.Parse(TXT_DefaultAlt.Text));
-
-                    }
-                }
-                catch (ArgumentException ex)
-                {
-                    CustomMessageBox.Show("Bad input options, please try again\n" + ex.ToString(), Strings.ERROR);
-                }
-
-                quickadd = false;
-                writeKML();
-            }
-        }
-
-        /// <summary>
-        /// Draw an mav icon, and update tracker location icon and guided mode wp on FP screen
-        /// </summary>
-        /// <param name="sender"></param>
-        /// <param name="e"></param>
-        public void timer1_Tick(object sender, EventArgs e)
-        {
-            try
-            {
-                if (isMouseDown || CurentRectMarker != null)
-                    return;
-
-                prop.alt = MainV2.comPort.MAV.cs.alt;
-                prop.altasl = MainV2.comPort.MAV.cs.altasl;
-                prop.center = MainMap.Position;
-                prop.Update(MainV2.comPort.MAV.cs.PlannedHomeLocation, MainV2.comPort.MAV.cs.Location,
-                    MainV2.comPort.MAV.cs.battery_kmleft);
-
-                // clear every 10 seconds
-                if(DateTime.Now.Second % 10 == 0)
-                    routesoverlay.Markers.Clear();
-
-                if (MainV2.comPort.MAV.cs.TrackerLocation != MainV2.comPort.MAV.cs.HomeLocation &&
-                    MainV2.comPort.MAV.cs.TrackerLocation.Lng != 0)
-                {
-                    addpolygonmarker(this, "Tracker Home", MainV2.comPort.MAV.cs.TrackerLocation.Lng,
-                        MainV2.comPort.MAV.cs.TrackerLocation.Lat, (int) MainV2.comPort.MAV.cs.TrackerLocation.Alt,
-                        Color.Blue, routesoverlay);
-                }
-
-                if (MainV2.comPort.MAV.cs.lat == 0 || MainV2.comPort.MAV.cs.lng == 0)
-                    return;
-
-                var marker = Common.getMAVMarker(MainV2.comPort.MAV, routesoverlay);
-
-                if(marker != null)
-                    routesoverlay.Markers.Add(marker);
-
-                if (MainV2.comPort.MAV.cs.mode.ToLower() == "guided" && MainV2.comPort.MAV.GuidedMode.x != 0)
-                {
-                    addpolygonmarker(this, "Guided Mode", MainV2.comPort.MAV.GuidedMode.y / 1e
```

Comment:
doesnt this just overwrite the start elements of the array populated in the previous line? it should have a destination index